input-output-hk/cardano-wallet-legacy
# [BIP44-239] Prepare to extend prefiltering to EOS wallets for ApplyBlock

https://github.com/input-output-hk/cardano-wallet/issues/239

<p align="right">#</p>

# Overview



- [x] In the BListener, I have extended prefiltering in`prefilterBlocks` to also prefilter for EO wallets during ApplyBlock
- [x] I have refactored prefilterBlock to return a single PrefilterResult, including a merged result in the case of prefiltering multiple wallet types together (i.e. there is at least one Eo and Fo wallet)
- [x] I have made the prefilter context for HdRnd (which was a list of `[(rootId,esk)]`) more consistent with Eo prefilter, a Map. Now both are maps.
- [x] I've changed `getWalletCredentials` to accept a snapshot instead of PassiveWallet (since the snapshot is already available at the call sites and this prevents a double getSnapshot)
- [x] I have moved `addrPoolGapByRootId` from the wallet layer to the Kernel, since I needed it to write `getEosWallets`. I also extended `addrPoolGapByRootId` to include accounts data along with the gap
- [x] I have written `getEosWallets`, that reconstructs Eo wallets and AddressPools from the AcidState db

# Comments



This PR adds prefiltering for Eo wallets during ApplyBlock. This means that we can now create new Eo wallets and track them during ApplyBlock.

NOTE: at this stage we cannot yet create new txs for Eo wallets. This will be in a subsequent commit.

# QA

We should test that we can create an Eo wallet, and if there are funds transferred to that wallet (using cardano=cli, say) we should see the funds appear in Daedelus.

We should also test Eo tracking in the presence of an HdRnd wallet! (In this case we have to merge prefilter results for the different wallet types, hence it needs special testing.
